gips/data/landsat/landsat.py

```diff
@@ -166,21 +180,27 @@ def __init__(self, filename):
 
         VerboseOut(("fname", fname), 2)
 
-        self.tile = fname[3:9]
-        year = fname[9:13]
-        doy = fname[13:16]
-        self.date = datetime.strptime(year + doy, "%Y%j")
-
         sr_pattern_re = re.compile(self._assets['SR']['pattern'])
         dn_pattern_re = re.compile(self._assets['DN']['pattern'])
+        c1_pattern_re = re.compile(self._assets['C1']['pattern'])
 
-        if sr_pattern_re.match(fname):
+        sr_match = sr_pattern_re.match(fname)
+        dn_match = dn_pattern_re.match(fname)
+        c1_match = c1_pattern_re.match(fname)
+
+        if sr_match:
             VerboseOut('SR asset', 2)
             self.asset = 'SR'
             self.sensor = 'LC8SR'
             self.version = int(fname[20:22])
-        elif dn_pattern_re.match(fname):
+        elif dn_match:
             VerboseOut('DN asset', 2)
+
+            self.tile = fname[3:9]
```

Comment:
optional:  It looks like you can use `dn_match.group(...)` calls as you did below.